### PR TITLE
Added tests for rendering the same Noise on different Level of detail

### PR DESCRIPTION
**Pull request by davids91**
Hi! 
 I thought there was some problem with rendering noisy data on different levels, because seemingly the crashes reacted to setting different resolution; So I wrote a test for that. 
 It turns out this isn't the case, as the crashes are because of incorrectly setting the transition sides somehow, so I'll work on it in another PR. 
 
 I hope you'll find this useful! 
 
 I'll be finalizing this in the coming days, meanwhile if you'll need any changes I'll be happy to discuss, otherwise I'm okay with it not being merged.

### CODE REVIEW

File: src/implementation/density_caching.rs [hunk 1]
```diff
@@ -205,42 +206,44 @@ where
...
+        debug_assert!(
+            x >= 0
+                && x <= subs as isize
+                && y >= 0
+                && y <= subs as isize
+                && z >= 0
+                && z <= subs as isize
+        );
+        let index = self.regular_block_index(x, y, z);
+        return self.regular_cache[index];
```
Comment (davids91) [inline src/implementation/density_caching.rs:246]: At this point the failure is obvious: a transition side is used where it shouldn't. 
Unfortunately I don't quite understand the failed assertion, but it if more  informative, than the previous. 
I would suggest adding a message here, e.g. like so: 
```suggestion
        debug_assert!(
            x >= 0
                && x <= subs as isize
                && y >= 0
                && y <= subs as isize
                && z >= 0
                && z <= subs as isize,
            "An incorrect transition side was added to an extraction!"
        );
        let index = self.regular_block_index(x, y, z);
        return self.regular_cache[index];
```

**Comment by Gnurfos** [comment]
Hi,

Thank you for the report and sorry if the bug is blocking you!

I don't agree 100% with your diagnostic, but this whole caching and density getting is a bit messy, it's not obvious (I'd like to refactor it but am low on time): getting density for transition cells tries to be smart and instead gets a regular point density, when this is possible. But regular points densities outside of the block are only loaded lazily, and too lazily (they should be loaded whenever we are about to need them, but it was done only in the context of a regular cell extraction). So here is a a case where a transition cell extraction needs a density from a regular grid point outside the block.

I pushed a rushed fix, and published 0.5.1

I'm sorry the PR got accidentally rejected when I renamed the main branch. I would welcome them if you can submit tests again illustrating the issue. However, when I ran them, they were very slow.

**Comment by Gnurfos** [comment]
I just added a test, which I think is the minimal case to reproduce the issue (but it doesn't do much in terms of making things less confusing).

**Comment by davids91** [comment]
I don't think I can add anything here :) Thank you for the fix!

**Pull request closed without merging**